src/redux/features/Plantings/plantingsSelectors.ts

```diff
@@ -1,11 +1,48 @@
 import { RootState } from 'src/redux/rootReducer';
+import { selectPlantingSites } from '../tracking/trackingSelectors';
+import { createSelector } from '@reduxjs/toolkit';
+import { PlantingSearchData } from './plantingsSlice';
 
 export const selectPlantings = (state: RootState) => state.plantings?.plantings;
 
+export const selectTotalPlantsBySubzone = (state: RootState) => {
+  const plantingsBySubzone: Record<number, { totalPlants: number }> = {};
+  state.plantings?.plantings?.forEach((planting) => {
+    if (plantingsBySubzone[Number(planting.plantingSubzone.id)]) {
+      plantingsBySubzone[Number(planting.plantingSubzone.id)] = {
+        totalPlants: plantingsBySubzone[Number(planting.plantingSubzone.id)].totalPlants + planting.totalPlants,
+      };
+    } else {
+      plantingsBySubzone[Number(planting.plantingSubzone.id)] = { totalPlants: planting.totalPlants };
+    }
+  });
+  return plantingsBySubzone;
+};
+
 export const selectPlantingsDateRange = (state: RootState, dateRange: string[]) =>
   selectPlantings(state)?.filter((planting) => {
     if (!dateRange || dateRange.length === 0) {
       return true;
     }
     return planting.createdTime > dateRange[0] && (dateRange.length < 2 || planting.createdTime < dateRange[1]);
   }) ?? [];
+
+export const selectPlantingProgressSubzones = createSelector(
+  [(state: RootState) => selectPlantingSites(state), (state: RootState) => selectTotalPlantsBySubzone(state)],
+  (plantingSites, plantingsBySubzone) => {
```

Comment:
this function will not be cached because plantingsBySubzone reference changes every time.. the code inside this function will be evalulated every time and can get slow.. you can fix this by moving selectTotalPlantsBySubzone into a cached selector, OR, make it an ordinary function that takes in the plantings (not state). Have this selector fetch both planting sites and plantings.. and call the getTotalPlantsBySubzone with the plantings.

```
  [ selectPlantingSites, selectPlantings],
  (plantingSites, plantings) => {
      const plantingsBySubzone = getTotalPlantsBySubzone(plantings);
      ... rest of your code
  }
```